engine/src/main/java/org/terasology/rendering/assets/texture/JPEGTextureFormat.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.rendering.assets.texture;
+
+import de.matthiasmann.twl.utils.PNGDecoder;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.assets.format.AbstractAssetFileFormat;
+import org.terasology.assets.format.AssetDataFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.file.PathMatcher;
+import java.util.List;
+
+/**
+ * @author Viveret (github.com/viveret)
+ */
+public class JPEGTextureFormat extends AbstractAssetFileFormat<TextureData> {
```

Comment:
This looks like copy and paste from `PNG`. If so, please adapt so they can both use a common method or even class without duplicating the code.